### DISCUSSION TITLE
Log or retrow rule

### DISCUSSION THREAD

**Discussion by svpiotrf** [Riverside-Software/sonar-openedge#1118] [General]
I have few code samples which I think are good and maybe there could be improved SonarQube rule to handle them:
1)
![image](https://github.com/Riverside-Software/sonar-openedge/assets/52634371/accbdca5-5920-4ab2-8504-115153d84f8d)
2)
![image](https://github.com/Riverside-Software/sonar-openedge/assets/52634371/1a03d14c-e587-4dab-bea2-55c88e42e2b2)
3)
![image](https://github.com/Riverside-Software/sonar-openedge/assets/52634371/4138958d-0232-46d7-a537-4dfa527ebceb)

And have question what would be best practice for this case:
![image](https://github.com/Riverside-Software/sonar-openedge/assets/52634371/816266d4-23c8-44ba-8d41-22289abc176e)

I not like suggestion to name variable to "uncaught" while it is against naming convention.
![image](https://github.com/Riverside-Software/sonar-openedge/assets/52634371/049dccbf-e167-40a1-a23a-294134fdd818)

And that "core" class for handle AppServer connections has no logger support and has no big sense to introduce it there.
Disconnect by my opinion should be silent, while it happens after action on AppServer were done and throwing an error may create bad situation. NO-ERROR option as well may overwrite original error with disconnect error.

**Comment by gquerret** [comment]
I agree that the rule triggers too many FP, that will be improved.  This rule also doesn't have a dedicated preprocessor directive to skip issues, that will be added.
